File: ruby-sdk/oneuptime/lib/oneuptime/timelineManager.rb
class OneUptimeTimelineManager
    def initialize(options)
        @options = options
        @timeLineStack = []
    end


    def addToTimeline(item)
        addItemToTimeline(item)
    end
    
    # return the timeline
    def getTimeline()
        return @timeLineStack
    end
    
    # clear the timeline
    def clearTimeline()
        @timeLineStack = []
    end 

   
    private
    def addItemToTimeline(item)
        # get the size of the stack
        if (@options[:maxTimeline] != nil && (@timeLineStack.length() == @options[:maxTimeline].to_i))
            return # It discards new timeline update once maximum is reached
        end

        # add time to it
        # current date and time
        time = Time.now
        now = time.inspect

        item["timestamp"] = now
        
        # add a new item to the stack
        @timeLineStack.append(item)
        return true
    end
end 

File: JavaScriptSDK/src/timelineManager.ts
class OneUptimeTimelineManager {
    private options: $TSFixMe;
    private timeLineStack: $TSFixMe;
    public constructor(options: $TSFixMe) {
        this.options = options;
        this.timeLineStack = [];
    }
    private _addItemToTimeline(item: $TSFixMe): void {
        // Get the size of the stack
        if (this.timeLineStack.length === this.options.maxTimeline) {
            // This.timeLineStack.shift(); // remove the oldest item
            return; // It discards new timline update once maximum is reached
        }
        // Add time to it
        item.timestamp = Date.now();
        // Add a new item to the stack
        this.timeLineStack.push(item);
        return true;
    }
    public addToTimeline(item: $TSFixMe): void {
        this._addItemToTimeline(item);
    }
    // Return the timeline
    public getTimeline(): void {
        return this.timeLineStack;
    }
    // Clear the timeline
    public clearTimeline(): void {
        this.timeLineStack = [];
    }
}
export default OneUptimeTimelineManager;
